code/go/0chain.net/chaincore/chain/state_handler.go

```diff
@@ -31,6 +34,46 @@ func SetupStateHandlers() {
 	http.HandleFunc("/_smart_contract_stats", common.UserRateLimit(c.SCStats))
 }
 
+func RespondSCRest(w http.ResponseWriter, r *http.Request, data interface{}, err error) {
+	if err != nil {
+		w.Header().Set("Content-Type", "application/json")
+		data := make(map[string]interface{}, 2)
+		data["error"] = err.Error()
+		if cErr, ok := err.(*common.Error); ok {
+			data["code"] = cErr.Code
+		}
+
+		switch {
+		case errors.Is(err, sc.NewErrInvalidRequest()):
+			w.WriteHeader(http.StatusBadRequest)
+		case errors.Is(err, sc.NewErrInternal()):
+			w.WriteHeader(http.StatusInternalServerError)
+		case errors.Is(err, sc.NewErrNoResource()):
+			w.WriteHeader(http.StatusNotFound)
+		default:
+			w.WriteHeader(http.StatusBadRequest)
+		}
+
+		buf := bytes.NewBuffer(nil)
+		json.NewEncoder(buf).Encode(data)
+		buf.WriteTo(w)
```

Comment:
Suggestion:
json.NewEncoder(w).Encode(data)

Reply:
Good suggestion, thank you.
I've made the appropriate changes.